src/mscorlib/shared/System/Threading/ReaderWriterLockSlim.cs

```diff
@@ -63,11 +63,17 @@ public class ReaderWriterLockSlim : IDisposable
         // the events that hang off this lock (eg writeEvent, readEvent upgradeEvent).
         private int _myLock;
 
+        // Used to deprioritize threading attempting to enter _myLock when they would not make progress after doing so
+        private int _enterMyLockDeprioritizationState;
```

Comment:
Please put a reasonably verbose comment about what this state is.   In particular it is a tuple of two counts (read and write), and these counts are requests to deprioritize that set of lock requestors).    All of this is hidden behind EnterSpinLock (I an sorely tempted to make a struct SpinLock which encapsulates _myLock and _enterMyLockDeprioritizationState and hides all this except for EnterMyLockSpin(EnterMyLockReason reason).  This keeps it conceptually localized).